vignettes/from-base.Rmd

```diff
@@ -8,41 +8,85 @@ vignette: >
-
-| base | stringr    |
-|------|------------|
-| `gregexpr(pattern, x)` | `str_locate_all(x, pattern)` |
-| `grep(pattern, x, value = TRUE)` | `str_subset(x, pattern)` |
-| `grep(pattern, x)` | `str_which(x, pattern)`  |
-| `grepl(pattern, x)` | `str_detect(x, pattern)` |
-| `gsub(pattern, replacement, x)` | `str_replace_all(x, pattern, replacement)`|
-| `nchar(x)` | `str_length(x)` |
-| `order(x)` | `str_order(x)` |
-| `regexec(pattern, x)` + `regmatches()` | `str_match(x, pattern)`  |
-| `regexpr(pattern, x)` + `regmatches()` | `str_extract(x, pattern)`|
-| `regexpr(pattern, x)` | `str_locate(x, pattern)`  |
-| `sort(x)` | `str_sort(x)` |
-| `strrep(x, n)` | `str_dup(x, n)` |
-| `strsplit(x, pattern)` | `str_split(x, pattern)`|
-| `strwrap(x)` | `str_wrap(x)` |
-| `sub(pattern, replacement, x)` | `str_replace(x, pattern, replacement)` |
-| `substr(x, start, end)` | `str_sub(x, start, end)` |
-| `tolower(x)` | `str_to_lower(x)` |
-| `tools::toTitleCase(x)` | `str_to_title(x)` |
-| `toupper(x)` | `str_to_upper(x)` |
-| `trimws(x)` | `str_trim(x)` |
+We'll begin with a lookup table between the most important stringr functions and their base R equivalents.
+
+```{r}
+#| label: stringr-base-r-diff
+#| echo: false
+
+data_stringr_base_diff <- tibble::tribble(
+  ~stringr,                                   ~`base_r`,
+  "str_detect(x, pattern)",                   "grepl(pattern, x)",
+  "str_dup(x, n)",                            "strrep(x, n)",
+  "str_extract(x, pattern)",                  "regmatches(x, regexpr(pattern, x))",
+  "str_extract_all(x, pattern)",              "regmatches(x, gregexpr(pattern, x))",
+  "str_length(x)",                            "nchar(x)",
+  "str_locate(x, pattern)",                   "regexpr(pattern, x)",
+  "str_locate_all(x, pattern)",               "gregexpr(pattern, x)",
+  "str_match(x, pattern)",                    "regmatches(x, regexec(pattern, x))",
+  "str_order(x)",                             "order(x)",
+  "str_replace(x, pattern, replacement)",     "sub(pattern, replacement, x)",
+  "str_replace_all(x, pattern, replacement)", "gsub(pattern, replacement, x)",
+  "str_sort(x)",                              "sort(x)",
+  "str_split(x, pattern)",                    "strsplit(x, pattern)",
+  "str_sub(x, start, end)",                   "substr(x, start, end)",
+  "str_subset(x, pattern)",                   "grep(pattern, x, value = TRUE)",
+  "str_to_lower(x)",                          "tolower(x)",
+  "str_to_title(x)",                          "tools::toTitleCase(x)",
+  "str_to_upper(x)",                          "toupper(x)",
+  "str_trim(x)",                              "trimws(x)",
+  "str_which(x, pattern)",                    "grep(pattern, x)",
+  "str_wrap(x)",                              "strwrap(x)"
+)
+
+# since downlit seems to omit tables, we manually link to pkgdown reference for HTML output
```

Comment:
I don't think this is the problem — I think the problem is that `library(stringr)` isn't shown in the output, so downlit can't see it. If you put `library(stringr)` in a separate chunk that's shown in the output, downlit should just work. This should substantially simplify the dependencies.

Reply:
I'm not familiar with how downlit works internally (it seems quite magic!). But the table cell content is still not autolinked even when there's a visible `library(stringr)` in the output:

<details>

![Screenshot 2022-12-06 at 22-44-49 From base R](https://user-images.githubusercontent.com/20040931/206042300-689113ee-25f6-44f7-a628-9b5ea1662b71.png)
</details>

I've tested a few combinations, none of them convinced downlit to autolink table cell contents. On the other hand, all code in text paragraphs is already properly linked **without** a visible `library(stringr)` chunk, so I think the issue lies elsewhere...

Reply:
Ok, in any case, I'd rather fix this upstream, rather than manually adding the links here.

Reply:
Would be great if this could be fixed in downlit. https://github.com/r-lib/downlit/issues/67 is about a different issue though, right?

I've only added the links in the stringr column since I think they're more important when consulting this table. But I wouldn't mind if both columns were autolinked. 😄

Do you want me to remove the custom downlit-code?

Reply:
Yes, please file a downlit issue, and remove the custom downlit code.